open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/validate/impl/VRPSubmissionPayloadValidator.java

```diff
@@ -0,0 +1,326 @@
+     * @param initiation The instruction initiation JSONObject from initiation request.
+     * @return A JSONObject indicating the validation result. It contains a boolean value under the key
+     *         ConsentExtensionConstants.IS_VALID_PAYLOAD, indicating whether the payload is valid. If the
+     *         validation fails, it returns a JSONObject containing error details with keys defined in ErrorConstants.
+     */
+    public static JSONObject validateInstruction(JSONObject submission,
+                                                JSONObject initiation) {
+
+        JSONObject validationResult = new JSONObject();
+        validationResult.put(ConsentExtensionConstants.IS_VALID_PAYLOAD, true);
+        validationResult.put(ConsentExtensionConstants.ERROR_CODE, "");
+        validationResult.put(ConsentExtensionConstants.ERROR_MESSAGE, "");
+
+        if (submission != null && initiation != null) {
+
+            if (submission.containsKey(ConsentExtensionConstants.INSTRUCTION_IDENTIFICATION)) {
+                Object value = submission.get(ConsentExtensionConstants.INSTRUCTION_IDENTIFICATION);
+                if (!(value instanceof String)) {
+                    return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_MISSING,
+                            ErrorConstants.INVALID_SUBMISSION_TYPE);
+                }
+            } else {
+                //log.error(ErrorConstants.INSTRUCTION_IDENTIFICATION_NOT_FOUND);
+                return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_MISSING,
+                        ErrorConstants.INSTRUCTION_IDENTIFICATION_NOT_FOUND);
+            }
+
+            if (submission.containsKey(ConsentExtensionConstants.END_TO_END_IDENTIFICATION)) {
+                Object value = submission.get(ConsentExtensionConstants.END_TO_END_IDENTIFICATION);
+                if (!(value instanceof String)) {
+                   // log.error(ErrorConstants.INVALID_TYPE);
+                    return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_MISSING,
+                            ErrorConstants.INVALID_TYPE);
+                }
+            } else {
+                return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_MISSING,
+                        ErrorConstants.END_TO_END_IDENTIFICATION_PARAMETER_NOT_FOUND);
+            }
+
+            //Validate Creditor Account
+            if (submission.containsKey(ConsentExtensionConstants.CREDITOR_ACC) &&
+                    initiation.containsKey(ConsentExtensionConstants.CREDITOR_ACC)) {
+
+                Object submissionCreditorAccounts = submission.get(ConsentExtensionConstants.CREDITOR_ACC);
+                Object consentInitiationCreditorAccounts = initiation.get(ConsentExtensionConstants.CREDITOR_ACC);
+
+                if (areJSONObjects(submissionCreditorAccounts, consentInitiationCreditorAccounts)) {
```

Comment:
`areJSONObjects` -> do we need to check these two elements at the same time whether these are instance of json objects? Else we can use isJsonObject and check only one at a time. WDYT?

Reply:
Fixed in [b6f317b]